apps/agent/contracts/Agent.sol

```diff
@@ -59,6 +137,28 @@ contract Agent is IERC165, ERC1271Bytes, IForwarder, IsContract, Vault {
         }
     }
 
+    /**
+    * @notice Add `_token.symbol(): string` to the list of protected tokens
+    * @param _token Address of the token to be protected
+    */
+    function addProtectedToken(address _token) external auth(ADD_PROTECTED_TOKEN_ROLE) {
+        require(protectedTokens.length < PROTECTED_TOKENS_CAP, ERROR_TOKENS_CAP_REACHED);
+        require(isContract(_token) || _token == ETH, ERROR_TOKEN_NOT_ETH_OR_CONTRACT);
```

Comment:
What if in addition to ensuring that `_token` is a contract, we perform a `balance(_token)` to make sure that the balance check doesn't revert for the token. If a token is added and `token.balanceOf()` reverts, it will block `safeExecute` until the token is removed from the protected list.

A malicious token could still be added that starts reverting after a state change or some period of time. To prevent this we could add a public function (without authentication) to remove a token that reverts, but this is a bit of an overkill IMO.

Reply:
> What if in addition to ensuring that _token is a contract, we perform a balance(_token) to make sure that the balance check doesn't revert for the token. If a token is added and token.balanceOf() reverts, it will block safeExecute until the token is removed from the protected list

Hmmmm. Yeah. I will add that too.

> A malicious token could still be added that starts reverting after a state change or some period of time. To prevent this we could add a public function (without authentication) to remove a token that reverts, but this is a bit of an overkill IMO.

Yeah I think it will mostly make things harder to understand [regarding roles and all] for end users while the treat is really low ...

Reply:
Thinking about it: what exactly is the threat model here?

1. If we are trying to deal with malicious tokens : what about the situation where calling `balanceOf` would actually empty your balance ? It's unlikely cause if the token is malicious / not trustless it could directly move your tokens [which would be pointless cause they would enp be worth nothing in the end].

2. We are just trying to prevent users to do a mistake while copy / pasting the address of the token ?

Reply:
For me the threat model is adding a contract to the list of protected tokens that can brick `safeExecute` until the token is removed. This could happen by mistake or maliciously and it could be specially bad if the permission to remove a token from the list is assigned to a long/complicated process or even burned.

But as you point out, adding a malicious token could bypass any check that we add here, so we would really just be protecting from mistakes.

> what about the situation where calling balanceOf would actually empty your balance ?

This shouldn't even be a problem since `balance(_token)` uses a `STATICCALL`.